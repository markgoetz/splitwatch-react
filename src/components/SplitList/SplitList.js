import React, { useContext } from 'react';
import TimerContext from '../../lib/TimerContext';
import Heading from '../Heading/Heading';
import renderTime from '../../lib/renderTime';
import styles from './SplitList.module.css';

export default function SplitList(props) {
    const splits = useContext(TimerContext).splits;

    const splitTimes = splits.map(split => (
        <li className={styles.split} key={split}>{renderTime(split)}</li>
    ));
    return (<div>
        <Heading level="2">Splits</Heading>
        <ol className={styles.list}>
            { splitTimes }
        </ol>
    </div>);
}
