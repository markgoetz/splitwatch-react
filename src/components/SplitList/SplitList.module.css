.list {
  list-style: decimal;
  margin-top: 10px;
  margin-left: 20px;
}

.split {
  font-family: var(--font-sans);
  font-size: 16px;
  padding: 5px 0;
}