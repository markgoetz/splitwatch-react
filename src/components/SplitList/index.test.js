import React from 'react';
import SplitList from './SplitList';
import { shallow } from 'preact-render-spy';

describe('SplitList', () => {
  test('renders correctly with no splits', () => {
    const context = shallow(<SplitList splits={[]} />);
    expect(context).toMatchSnapshot();
  });

  test('renders correctly with splits', () => {
    const context = shallow(<SplitList splits={[1, 2, 3, 4, 5]} />);
    expect(context).toMatchSnapshot();
  });
});