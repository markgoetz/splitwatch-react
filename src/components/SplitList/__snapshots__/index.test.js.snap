// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitList renders correctly with no splits 1`] = `
preact-render-spy (1 nodes)
-------
<div>
  <Heading level="2">Splits</Heading>
  <ol></ol>
</div>

`;

exports[`SplitList renders correctly with splits 1`] = `
preact-render-spy (1 nodes)
-------
<div>
  <Heading level="2">Splits</Heading>
  <ol>
    <li>00:00:00.0</li>
    <li>00:00:00.0</li>
    <li>00:00:00.0</li>
    <li>00:00:00.0</li>
    <li>00:00:00.0</li>
  </ol>
</div>

`;
